modules/xmpp/xmpp.js

```diff
@@ -92,7 +92,11 @@ export default class XMPP extends Listenable {
         // they wanted to utilize the connected connection in an unload handler
         // of their own. However, it should be fairly easy for them to do that
         // by registering their unload handler before us.
-        $(window).on('beforeunload unload', this.disconnect.bind(this));
+        $(window).on('beforeunload unload', ev => {
+            this.disconnect(ev).catch(() => {
+                // ignore errors in order to not brake the unload.
+            });
```

Comment:
To confirm, uncaught promise rejections somehow prevent other unload logic from executing?

Reply:
Yep. Noticed this on Electron! very strange...